Assets/MixedRealityToolkit-Preview/CameraCapture/CameraCaptureUWP.cs

```diff
@@ -0,0 +1,230 @@
+			});
+		}
+		/// <summary>
+		/// Request an image from the camera, and provide it as a GPU Texture!
+		/// </summary>
+		/// <param name="onImageAcquired">This is the function that will be called when the image is ready. Texture is not guaranteed to be a Texture2D, could also be a WebcamTexture. Matrix is the transform of the device when the picture was taken.</param>
+		public void RequestImage(Action<Texture, Matrix4x4> aOnImageAcquired)
+		{
+			if (!isReady || IsRequestingImage)
+				return;
+
+			GetImage((tex, transform) => {
+				if (aOnImageAcquired != null)
+					aOnImageAcquired(tex, transform);
+			});
+		}
+
+		/// <summary>
+		/// Done with the camera, free up resources!
+		/// </summary>
+		public void Shutdown()
+		{
+			if (cacheTex   != null) GameObject.Destroy(cacheTex);
+			if (resizedTex != null) GameObject.Destroy(resizedTex);
+
+			if (camera != null)
+				camera.Dispose();
+			camera = null;
+		}
+		#endregion
+	}
+
+	public sealed class VideoDeviceControllerWrapper : IDisposable
+	{
+		private VideoDeviceController controller = null;
+
+		public VideoDeviceControllerWrapper(IntPtr unknown)
+		{
+			controller = (VideoDeviceController)Marshal.GetObjectForIUnknown(unknown);
+		}
+ 
+		~VideoDeviceControllerWrapper()
+		{
+			Dispose();
+		}
+ 
+		public void SetExposure(int exposure)
+		{
+			//Debug.LogFormat("({3} : {0}-{1}) +{2}", controller.Exposure.Capabilities.Min, controller.Exposure.Capabilities.Max, controller.Exposure.Capabilities.Step, exposure);
+			if (!controller.Exposure.TrySetAuto(false))
+				Debug.LogErrorFormat("[{0}] HoloLens locatable camera has failed to set auto exposure off", typeof(VideoDeviceControllerWrapper));
+
+			if (!controller.Exposure.TrySetValue((double)exposure))
+				Debug.LogErrorFormat("[{0}] HoloLens locatable camera has failed to set exposure to {1} as requested", typeof(VideoDeviceControllerWrapper), exposure);
+		}
+		public void SetWhiteBalance(int kelvin)
+		{
+			if (!controller.WhiteBalance.TrySetAuto(false))
+				Debug.LogErrorFormat("[{0}] HoloLens locatable camera has failed to set auto WhiteBalance off", typeof(VideoDeviceControllerWrapper));
+
+			if (!controller.WhiteBalance.TrySetValue((double)kelvin))
+				Debug.LogErrorFormat("[{0}] HoloLens locatable camera has failed to set WhiteBalance to {1} as requested", typeof(VideoDeviceControllerWrapper), kelvin);
+		}
+		public void SetFocus(int focus)
+		{
+			if (!controller.Focus.TrySetAuto(false))
+				Debug.LogErrorFormat("[{0}] HoloLens locatable camera has failed to set auto Focus off", typeof(VideoDeviceControllerWrapper));
+
+			if (!controller.Focus.TrySetValue((double)focus))
+				Debug.LogErrorFormat("[{0}] HoloLens locatable camera has failed to set Focus to {1} as requested", typeof(VideoDeviceControllerWrapper), focus);
+		}
+		public void SetISO(int iso)
+		{
+			var task = controller.IsoSpeedControl.SetValueAsync((uint)iso);
+		}
+ 
+		public void Dispose()
```

Comment:
you should call GC.SuppressFinalize here.  please see https://blogs.msdn.microsoft.com/davidklinems/2006/01/10/wrapping-unmanaged-resources/ for an example dispose pattern implementation